### PR TITLE
Add enum declaration support to new parser

### LINKED ISSUE #23414: Add enum declaration support to new parser
**Description:**
https://ballerina.io/ballerina-spec/spec.html#section_8.8

**Suggested Labels (optional):**
- Component/Parser
- Type/NewFeature

**Suggested Assignees (optional):**
- @rdulmina

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
Add enum declaration support to new parser with recovery logic and test cases.

Fixes #23414

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by rdulmina** [comment]
Please ignore the branch name.

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -10857,4 +10866,164 @@ private STNode parseConditionalExpression(STNode lhsExpr) {
...
+        }
+
+        return STNodeFactory.createEnumMemberNode(metadata, identifierNode, equalToken, constExprNode);
+    }
+
+    private STNode parseEnumMemberRhs() {
+        return parseEnumMemberRhs(peek().kind);
+    }
+
+    private STNode parseEnumMemberRhs(SyntaxKind nextTokenKind) {
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:10995]: Can say `parseEnumMemberEnd`

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 2]
```diff
@@ -10857,4 +10866,164 @@ private STNode parseConditionalExpression(STNode lhsExpr) {
...
+
+        STNode identifierNode = parseIdentifier(ParserRuleContext.ENUM_MEMBER_NAME);
+        return parseEnumMemberInternalRhs(metadata, identifierNode);
+    }
+
+    private STNode parseEnumMemberInternalRhs(STNode metadata, STNode identifierNode) {
+        return parseEnumMemberInternalRhs(metadata, identifierNode, peek().kind);
+    }
+
+    private STNode parseEnumMemberInternalRhs(STNode metadata, STNode identifierNode, SyntaxKind nextToken) {
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:10962]: can say `parseEnumMemberRhs`

Comment (SupunS) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:10962]: Let's have the `SyntaxKind nextToken` as the first param. Thats the pattern we have followed elsewhere...

**Pull request merged**